pkg/client/client.go

```diff
@@ -1,75 +1,439 @@
+	}
+
+	// If kubeconfig bytes are provided, load directly from bytes
+	if len(kubeconfigFile) != 0 {
+		config, err = clientcmd.RESTConfigFromKubeConfig(kubeconfigFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build Kubernetes config from provided kubeconfig: %w", err)
+		}
+		l.Debug("Created Kubernetes config from provided kubeconfig")
+	} else {
+		// Fall back to default kubeconfig file
+		var kubeconfig string
+		if home := homedir.HomeDir(); home != "" {
+			// Default kubeconfig file
+			kubeconfig = filepath.Join(home, ".kube", "config")
+		}
+
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build Kubernetes config: %w", err)
+		}
+		l.Debug("Created Kubernetes config from default kubeconfig file")
+	}
+	httpClient, err := getUhttpClient(ctx, config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to create HTTP client: %w", err)
+	}
+	clientset, err := kubernetes.NewForConfigAndClient(config, httpClient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Kubernetes clientset: %w", err)
+	}
+	return clientset, nil
 }
 
-// GetAccounts fetches a list of real accounts from ArgoCD using the CLI.
-// Command: argocd account list --output json.
+// ensureAuthenticated ensures the client is authenticated with ArgoCD API.
+// It authenticates using the /api/v1/session endpoint and stores the session token.
+func (c *Client) ensureAuthenticated(ctx context.Context) error {
+	l := ctxzap.Extract(ctx)
+
+	// If we already have a session token, try to verify it
+	if c.sessionToken != "" {
+		if err := c.verifySession(ctx); err == nil {
+			l.Debug("ArgoCD API session token still valid")
+			return nil
+		} else {
+			l.Debug("ArgoCD API session token expired or invalid", zap.Error(err))
+			c.sessionToken = ""
+		}
+	}
+
+	// Authenticate with ArgoCD API
+	l.Debug("Authenticating with ArgoCD API")
+	authURL, err := c.buildURL(sessionURL)
+	if err != nil {
+		return fmt.Errorf("failed to build session URL: %w", err)
+	}
+
+	loginData := map[string]string{
+		"username": c.username,
+		"password": c.password,
+	}
+
+	jsonData, err := json.Marshal(loginData)
+	if err != nil {
+		return fmt.Errorf("failed to marshal login data: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, authURL, strings.NewReader(string(jsonData)))
+	if err != nil {
+		return fmt.Errorf("failed to create login request: %w", err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to authenticate with ArgoCD API: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		bodyBytes, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("authentication failed with status %d: %s", resp.StatusCode, string(bodyBytes))
+	}
+
+	// Extract session token from Set-Cookie header
+	// ArgoCD typically sets a cookie named "argocd.token"
```

Comment:
So we don't tell it which auth to use? It just picks one of the two? Is this a difference in the version of argocd or how the user configured argocd?